### PR TITLE
Protocol lib

### PR DESCRIPTION
**Pull request by CJCrafter**
Finally address #111, a rare issue on paper servers that effects an extremely small minority of users. Unfortunately, in order to solve this problem we had out outsource our packet listener to ProtocolLib. MechanicsCore now required ProtocolLib to be installed. This is a small price to pay, since most servers already use ProtocolLib, and it is fairly stable across versions. 

We need to create an auto-downloader though. Asking people to download ProtocolLib will not be effective, as people love to leave bad reviews with ah "doesn't work" message the first chance they get.

### CODE REVIEW

**Comment by CJCrafter** [comment]
Currently, the latest release of ProtocolLib does not support 1.19. They have a dev build, but since *most* users don't use the development build, we will wait to merge this branch until that happens

**Review by DeeCaaD — DISMISSED** [review]

File: WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/packetlisteners/OutAbilitiesListener.java [hunk 1]
```diff
@@ -1,46 +1,41 @@
...
     @Override
-    public void onPacket(Packet packet) {
-        EntityWrapper entityWrapper = WeaponMechanics.getEntityWrapper(packet.getPlayer());
+    public void onPacketSending(PacketEvent event) {
+        EntityWrapper entity = WeaponMechanics.getEntityWrapper(event.getPlayer());
 
-        ZoomData main = entityWrapper.getMainHandData().getZoomData();
-        ZoomData off = entityWrapper.getOffHandData().getZoomData();
+        ZoomData main = entity.getMainHandData().getZoomData();
+        ZoomData off = entity.getOffHandData().getZoomData();
```
Comment (nicosammito) [inline WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/packetlisteners/OutAbilitiesListener.java:27]:
```suggestion
        ZoomData mainHandZoomData = entity.getMainHandData().getZoomData();
        ZoomData offHandZoomData = entity.getOffHandData().getZoomData();
```

better naming

File: WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/packetlisteners/OutAbilitiesListener.java [hunk 2]
```diff
@@ -1,46 +1,41 @@
...
 
-        // Not zooming
-        if (!main.isZooming() && !off.isZooming()) return;
+        // Player is not scoped in, no need to modify their FOV
+        if (!main.isZooming() && !off.isZooming())
+            return;
```
Comment (nicosammito) [inline WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/packetlisteners/OutAbilitiesListener.java:31]:
```suggestion
        // Player is not scoped in, no need to modify their FOV
        if (!mainHandZoomData.isZooming() && !offHandZoomData.isZooming()) return;
```

File: WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/packetlisteners/OutAbilitiesListener.java [hunk 3]
```diff
@@ -1,46 +1,41 @@
...
 
         double zoomAmount = main.isZooming() ? main.getZoomAmount() : off.getZoomAmount();
```
Comment (nicosammito) [inline WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/packetlisteners/OutAbilitiesListener.java:33]:
```suggestion
        double zoomAmount = mainHandZoomData.isZooming() ? mainHandZoomData.getZoomAmount() : offHandZoomData.getZoomAmount();
```

**Review by nicosammito — COMMENTED** [review]
just some small changes

**Pull request merged**